src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -373,7 +451,7 @@ private Expression ProjectElement(Expression source, SelectExpandClause selectEx
 
         private static bool GetSelectsOpenTypeSegments(SelectExpandClause selectExpandClause, IEdmEntityType entityType)
         {
-            if (!entityType.IsOpen)
+            if (entityType == null || !entityType.IsOpen)
```

Comment:
Why not support open complex types?

Reply:
I wanted to limit the scope but I guess I will now. TODO: Add a test case. 

---
In reply to: [308506219](https://github.com/OData/WebApi/pull/1738#discussion_r308506219) [](ancestors = 308506219)

Reply:
Did you add the test case?

---
In reply to: [326413640](https://github.com/OData/WebApi/pull/1738#discussion_r326413640) [](ancestors = 326413640,308506219)